SetupDataPkg/Library/ConfigVariableListLib/ConfigVariableListLib.c

```diff
@@ -0,0 +1,316 @@
+/** @file
+  Library interface to process the list of configuration variables.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/DebugLib.h>
+#include <PiDxe.h>
+#include <Library/DxeServicesLib.h>
+#include <Library/ConfigVariableListLib.h>
+
+/**
+  Parse Active Config Variable List and return full list or specific entry if VarName parameter != NULL
+
+  @param[out] ConfigVarListPtr    Pointer to configuration data. User is responsible to free the
+                                  returned buffer and the Data, Name fields for each entry.
+  @param[out] ConfigVarListCount  Number of variable list entries.
+  @param[in]  ConfigVarName       If NULL, return full list, else return entry for that variable
+
+  @retval EFI_INVALID_PARAMETER   Input argument is null.
+  @retval EFI_OUT_OF_RESOURCES    Memory allocation failed.
+  @retval EFI_NOT_FOUND           The requested variable is not found in the active profile.
+  @retval EFI_COMPROMISED_DATA    The profile contains data that does not fit within the structure defined.
+  @retval EFI_SUCCESS             The operation succeeds.
+
+**/
+EFI_STATUS
+ParseActiveConfigVarList (
+  OUT CONFIG_VAR_LIST_ENTRY  **ConfigVarListPtr,
+  OUT UINTN                  *ConfigVarListCount,
+  IN  CONST CHAR16           *ConfigVarName
+  )
+{
+  UINTN                  BinSize  = 0;
+  CONFIG_VAR_LIST_HDR    *VarList = NULL;
+  EFI_STATUS             Status   = EFI_SUCCESS;
+  CHAR16                 *VarName = NULL;
+  CHAR8                  *Data    = NULL;
+  CHAR16                 *NameInBin;
+  EFI_GUID               *Guid;
+  UINT32                 Attributes;
+  CHAR8                  *DataInBin;
+  UINT32                 CRC32;
+  UINT32                 ListIndex = 0;
+  CONFIG_VAR_LIST_ENTRY  *Entry;
+  VOID                   *BlobPtr = NULL;
+
+  if (!ConfigVarListPtr || !ConfigVarListCount) {

```

Comment:
nit: we normally do comparison as:
```suggestion
  if ((ConfigVarListPtr == NULL) || (ConfigVarListCount == NULL)) {
```

Reply:
Changed where I saw it.